AK/Optional.h

```diff
@@ -55,16 +59,37 @@ requires(!IsLvalueReference<T>) class [[nodiscard]] Optional<T> {
 
     static_assert(!IsLvalueReference<T> && !IsRvalueReference<T>);
 
+    // FIXME: GCC seems to have an issue with uninitialized unions
+    //        which forces us to have an equally sized trivial null member in the union
+    //        to pseudo-initialize the union,
+    //        Note this could/should be an AK::Array, but it can't as that depends on Optional,
+    //        we use an anonymous struct with a dummy member instead
+    constexpr ALWAYS_INLINE void construct_null()
+    {
+#ifdef AK_COMPILER_GCC
+        m_null = {};
+#endif
+    }
+
 public:
     using ValueType = T;
 
-    ALWAYS_INLINE Optional() = default;
+    // FIXME: Why do we need the IsConst here, probably fixed in c++26
```

Comment:
I _think_ giving that "null" member a default ctor will allow you to delete this ctor entirely, not sure though.

Reply:
For some reason we still need that ctor, but I was able to broaden the requirement on it
and added a Test for superstition sake